### PR TITLE
add auc loss measure (area under ROC curve)

### PR DESCRIPTION
**Pull request by Prometheus77**

### CODE REVIEW

File: src/scicloj/metamorph/ml/loss.clj [hunk 1]
```diff
@@ -57,3 +57,22 @@ that label."
...
+
+
+(defn auc
+  "Calculates area under the ROC curve. Uses AUC formula from R's 'mlr' package.
+  (sum(r[i]) - n.pos * (n.pos + 1) / 2) / (n.pos * n.neg)
+  See https://github.com/mlr-org/mlr/blob/main/R/measures.R"
+  ^double [predictions labels]
+  (assert (= (count predictions) (count labels)))
+  (assert (= (set labels) #{0 1}))
+  (let [eval-data (ds/dataset {:predictions predictions
```
Comment (behrica) [inline src/scicloj/metamorph/ml/loss.clj:69]: Alias `ds` is not defined in this name space.

File: src/scicloj/metamorph/ml/loss.clj [hunk 2]
```diff
@@ -57,3 +57,22 @@ that label."
...
+                               :labels labels})
+        ranked-data (ds/order-by eval-data :predictions :desc)
```
Comment (behrica) [inline src/scicloj/metamorph/ml/loss.clj:71]: order-by is not defined on `tech.ml.dataset` but on `tablecoth`.
So please use `tc` as alias

**Comment by behrica** [comment]
Could you as well add a test in a (new) test file  "loss_test.clj" in the "test" folder ?

Thanks for you PR !

**Comment by Prometheus77** [comment]
I created a separate pull request with an improved function and test

**Pull request closed without merging**